### PR TITLE
Add DB Schema

### PR DESCRIPTION
**Pull request by kratsg**
```
2017-10-25T00:32:13.635925+00:00 app[api]: Deploy d76910e9 by user kratsg@gmail.com
2017-10-25T00:32:21.827208+00:00 app[web.1]: signsfive-api listening at http://[::]:54396
2017-10-25T00:32:21.989604+00:00 app[web.1]: Executing (default): SELECT 1+1 AS result
2017-10-25T00:32:21.999789+00:00 app[web.1]: {"name":"signsfive-api","hostname":"bb0310f8-0ef6-47f3-9534-c94b3fa172c9","pid":18,"level":30,"msg":"Database connection has been established successfully.","time":"2017-10-25T00:32:21.999Z","v":0}
2017-10-25T00:32:22.138772+00:00 heroku[web.1]: State changed from starting to up
```

we can definitely authenticate to the database correctly. This is a WIP and I should add the rest of the functionality soon.

### CODE REVIEW

**Comment by kratsg** [comment]
Progress so far:

- all models added
- reorganized code to make database definition look cleaner

Next:

- correctly define associations for our models based on the lingo here: http://docs.sequelizejs.com/manual/tutorial/associations.html 
- complete validation definition of our models based on: http://docs.sequelizejs.com/manual/tutorial/models-definition.html#validations

File: db/models/sign.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+'use strict';
+module.exports = (sequelize, DataTypes) => {
+  var Sign = sequelize.define('Sign', {
+    name: DataTypes.STRING,
```
Comment (meltedspork) [inline db/models/sign.js]: Do we need `id` in the sign model?

File: db/models/terminology.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+module.exports = (sequelize, DataTypes) => {
+  var Terminology = sequelize.define('Terminology', {
+    name: DataTypes.STRING,
```
Comment (meltedspork) [inline db/models/terminology.js]: Do we need id in the terminology model?

File: db/models/user.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+'use strict';
+module.exports = (sequelize, DataTypes) => {
+  var User = sequelize.define('User', {
+    email: {
```
Comment (meltedspork) [inline db/models/user.js]: Do we need id in the user model?

File: db/models/introspect.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+module.exports = (sequelize, DataTypes) => {
+  var Introspect = sequelize.define('Introspect', {
+    name: DataTypes.STRING,
```
Comment (meltedspork) [inline db/models/introspect.js]: Do we need id in the introspect model?

File: db/models/index.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+var fs        = require('fs');
```
Comment (meltedspork) [inline db/models/index.js]: How do you feel that in new (maybe next?) PR that we could use babel here so we can use ES2015+?

File: db/models/introspect.js (hunk 4 above)
Comment (kratsg) [reply db/models/introspect.js]: Sequelize will add this automatically when sync'ing the table (along with created/updated fields). We can be more explicit about this.

File: db/models/sign.js (hunk 1 above)
Comment (kratsg) [reply db/models/sign.js]: Sequelize will add this automatically when sync'ing the table (along with created/updated fields). We can be more explicit about this.

File: db/models/terminology.js (hunk 2 above)
Comment (kratsg) [reply db/models/terminology.js]: Sequelize will add this automatically when sync'ing the table (along with created/updated fields). We can be more explicit about this.

File: db/models/user.js (hunk 3 above)
Comment (kratsg) [reply db/models/user.js]: Sequelize will add this automatically when sync'ing the table (along with created/updated fields). We can be more explicit about this.

**Comment by kratsg** [comment]
> How do you feel that in new (maybe next?) PR that we could use babel here so we can use ES2015+?

Sure. Most of the files were generated by the sequelize-cli which just didn't directly support ES2015+ out of the box. For now, just to make sure things worked, I just did everything here in vanilla javascript with the ulterior goal to migrate.

**Comment by meltedspork** [comment]
> Sure. Most of the files were generated by the sequelize-cli which just didn't directly support ES2015+ out of the box. For now, just to make sure things worked, I just did everything here in vanilla javascript with the ulterior goal to migrate.

That makes sense. We can keep everything vanilla JavaScript for sequelize

**Comment by kratsg** [comment]
Relationship overview:

```
Source.B2M(Gloss, ...)
Gloss.B2M(Source, ...)
Source.B2M(Cat, ...)
Sign.B2M(Cat, ...)
Cat.B2M(Source, ...)
Sign.B2M(Gloss, ...)
Sign.B2M(Region, ...)
Cat.B2M(Sign, ...)
Gloss.B2M(Gloss, gloss1)
Gloss.B2M(Gloss, gloss2)
Regions.B2M(Sign, ...)
Author.hasMany(Sign, ...) [author ID added to sign table]
Regions.hasMany(Author, ...) [region ID added to author table]
```

**Comment by meltedspork** [comment]
Lets fix the naming tables a bit with grammar -

Authors
Signs
Regions
Sources
Categories
Glosses

**Comment by kratsg** [comment]
SQL generated so far:

```
Executing (default): CREATE TABLE IF NOT EXISTS `Authors` (`id` INTEGER NOT NULL auto_increment , `email` VARCHAR(255), `username` VARCHAR(255), `auth0_id` VARCHAR(255), `createdAt` DATETIME NOT NULL, `updatedAt` DATETIME NOT NULL, PRIMARY KEY (`id`)) ENGINE=InnoDB;
Executing (default): SHOW INDEX FROM `Authors` FROM `signsfive_dev`
Executing (default): DROP TABLE IF EXISTS `Categories`;
Executing (default): CREATE TABLE IF NOT EXISTS `Categories` (`id` INTEGER NOT NULL auto_increment , `name` VARCHAR(255), `description` TEXT, `createdAt` DATETIME NOT NULL, `updatedAt` DATETIME NOT NULL, PRIMARY KEY (`id`)) ENGINE=InnoDB;
Executing (default): SHOW INDEX FROM `Categories` FROM `signsfive_dev`
Executing (default): DROP TABLE IF EXISTS `Glosses`;
Executing (default): CREATE TABLE IF NOT EXISTS `Glosses` (`id` INTEGER NOT NULL auto_increment , `name` VARCHAR(255), `description` TEXT, `createdAt` DATETIME NOT NULL, `updatedAt` DATETIME NOT NULL, PRIMARY KEY (`id`)) ENGINE=InnoDB;
Executing (default): SHOW INDEX FROM `Glosses` FROM `signsfive_dev`
Executing (default): DROP TABLE IF EXISTS `Regions`;
Executing (default): CREATE TABLE IF NOT EXISTS `Regions` (`id` INTEGER NOT NULL auto_increment , `name` VARCHAR(255), `loc` POINT, `createdAt` DATETIME NOT NULL, `updatedAt` DATETIME NOT NULL, PRIMARY KEY (`id`)) ENGINE=InnoDB;
Executing (default): SHOW INDEX FROM `Regions` FROM `signsfive_dev`
Executing (default): DROP TABLE IF EXISTS `Signs`;
Executing (default): CREATE TABLE IF NOT EXISTS `Signs` (`id` INTEGER NOT NULL auto_increment , `giphy_id` VARCHAR(255), `createdAt` DATETIME NOT NULL, `updatedAt` DATETIME NOT NULL, PRIMARY KEY (`id`)) ENGINE=InnoDB;
Executing (default): SHOW INDEX FROM `Signs` FROM `signsfive_dev`
Executing (default): DROP TABLE IF EXISTS `Sources`;
Executing (default): CREATE TABLE IF NOT EXISTS `Sources` (`id` INTEGER NOT NULL auto_increment , `url` VARCHAR(255), `createdAt` DATETIME NOT NULL, `updatedAt` DATETIME NOT NULL, PRIMARY KEY (`id`)) ENGINE=InnoDB;
Executing (default): SHOW INDEX FROM `Sources` FROM `signsfive_dev`
```

I will encode this into a migration soon.

**Comment by kratsg** [comment]
signsfive_dev_schema

**Comment by kratsg** [comment]
signsfive_dev_schema

**Comment by kratsg** [comment]
To do:
- determine if we should use scopes for `sourcable` models: gloss and sign. Could potentially combine into a single table with a foreign key that's joined on another table depending on the value of a column [see http://docs.sequelizejs.com/manual/tutorial/associations.html#scopes]
- fix up primary keys for relationship-relationship tables, specifically they should not have an `id` that's auto-incrementing, but a combined primary key -- fix up the migration here since we do not have a database yet.

**Comment by kratsg** [comment]
signsfive_dev_schema

**Comment by kratsg** [comment]
@meltedspork time for you to look things over!

**Comment by meltedspork** [comment]
SWEET!!!!

**Comment by kratsg** [comment]
@meltedspork approved offline.

**Pull request merged**